src/Tribe/Assets.php

```diff
@@ -38,6 +38,45 @@ public static function instance() {
 	public function __construct() {
 		// Hook the actual registering of.
 		add_action( 'init', [ $this, 'register_in_wp' ], 1, 0 );
+		add_filter( 'script_loader_tag', [ $this, 'filter_tag_async_defer'], 50, 2 );
+	}
+
+	/**
+	 * Filters the Script tags to attach Async and/or Defer based on the rules we set in our Asset class.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $tag    Tag we are filtering.
+	 * @param string $handle Which is the ID/Handle of the tag we are about to print.
+	 *
+	 * @return string Script tag with the defer and/or async attached.
+	 */
+	public function filter_tag_async_defer( $tag, $handle ) {
+		// Only filter for own own filters.
```

Comment:
Can we put a sanity check in here to be sure it's a script tag - it's a public function, so someone may call it incorrectly (and a script and style asset _might_ be named the same)?